### ISSUE TITLE
Csub Doesn't properly handle my pipe |

### ISSUE THREAD

**Issue by SZRabinowitz** [hollasch/csub#12]
Problem: I tried running the command:
```
csub adb pull `adb shell "pm path com.android.packageinstaller | sed s/package://g"`
```

It returned: 
```
'sed' is not recognized as an internal or external command,
operable program or batch file.
adb.exe: pull requires an argument
```

What I expected: 
I expected it to run everything inside the backticks, that is:
`adb shell "pm path com.android.packageinstaller | sed s/package://g"`, and use the ouptut of it in the full command. 

When I run the command manually: 
```
C:\Users\Admin\Downloads>adb shell "pm path com.android.packageinstaller | sed s/package://g"
/system/priv-app/PackageInstaller/PackageInstaller.apk
```

Now I can run `adb pull /system/priv-app/PackageInstaller/PackageInstaller.apk` and have no errors:
```
/system/priv-app/PackageInstaller/PackageInstall..., 0 skipped. 14.4 MB/s (2803184 bytes in 0.185s)
```

As you see, I ran adb shell then put quotes around the shell command, so sed is ran inside of adb shell, not inside of windows. But csub doesnt respect that properly. 

Here is another example of an issue with pipes:
```
C:\Users\Admin\Downloads>dir | findstr test.txt
11/01/2023  06:02 PM           120,606 test.txt

C:\Users\Admin\Downloads>csub echo `dir | findstr test.txt`
Error:  Mismatched ` quotes.
```

**Comment by SZRabinowitz** [comment]
after struggling with this for days, I found a solution minutes after I created this issue. 


adding a caret before the pipe solved it.

for example:
```
C:\Users\Admin\Downloads>csub adb pull `adb shell "pm path com.android.packageinstaller ^| sed s/package://g"`
/system/priv-app/PackageInstaller/PackageInstall..., 0 skipped. 14.5 MB/s (2803184 bytes in 0.185s)
```
and also 
```
C:\Users\Admin\Downloads>csub echo `dir ^| findstr test.txt`
11/01/2023  06:02 PM           120,606 test.txt
```


I will let you decide whether to close this issue or not, because I think it would still preferable for csub to pass all special characters to the command being substituted - I would prefer not to escape. so really any characters between the \`backticks\` should all be part of the command getting substituted, but I don't know whether that's possible or how hard it would be to implement. 


Anyways, I appreciate this great tool, it makes it easier for me to not need to use confusing for commands whenever I need to substitute output.

**Comment by hollasch** [comment]
The essential problem here is that `csub` should really be implemented in the shell (or terminal), but we're faking it here. The standard shells know how to deal with double-quoted strings, for example, but not with back ticks. So rather than passing the entire expression `dir | findstr test.txt` to `csub`, instead we get individual arguments like `` `dir `` and `` test.txt` ``. As you can see, the back ticks are not interpreted specially by the shell, which leaves the pipe symbol to be seen and interpreted by the shell, which is incorrect. All of this unintended handling by the shell happens before the `csub` command even _starts_, so by the time the command is running, it's already too late.

As you've noticed, you can escape the pipe symbol to tell the shell that it is not to be interpreted. In addition, since the shell understands double-quotes enough to keep from trying to interpret the contents, you can also enclose the entire back tick expression inside double quotes, like so:

    echo "`dir | findstr test.txt`"

This is also nice because you can pass multiple pipe symbols (and other special terminal symbols) without having them be interpreted by the shell.

Though I can't eliminate the need to escape or quote, I can add a bit more information to the command help. I'll do that shortly.

**Comment by SZRabinowitz** [comment]
Thanks, I appreciate that information very much, and I think this issue can be closed (unless you want to wait until you make the new commit).

Gonna star this repo!!!

**Comment by hollasch** [comment]
Fixed; release v1.2.1.
